### PR TITLE
feat(agent): add wait until into builder

### PR DESCRIPTION
**Pull request by nandor-magyar**
Additionally, I've refactored the relevant codeparts where this was already in use.

### CODE REVIEW

**Review by c3ppc3pp — APPROVED** [review]
We talked about this personally in detail,
LGTM

File: agent/pkg/dagent/utils/docker.go [hunk 1]
```diff
@@ -405,9 +406,12 @@ func WithImportContainer(dc *containerbuilder.DockerContainerBuilder, importConf
 			mountList []mount.Mount,
 			logger *io.StringWriter) error {
 			if initError := spawnInitContainer(client, ctx, containerName, mountList, importConfig, dog, cfg); initError != nil {
-				log.Printf("Failed to spawn init container: %v", initError)
+				errStr := fmt.Sprintf("Failed to spawn init container: %v", initError)
+				log.Println(errStr)
```
Comment (polaroi8d) [inline agent/pkg/dagent/utils/docker.go:410]: log.Println?

File: agent/pkg/dagent/utils/docker.go [hunk 2]
```diff
@@ -405,9 +406,12 @@ func WithImportContainer(dc *containerbuilder.DockerContainerBuilder, importConf
 			mountList []mount.Mount,
 			logger *io.StringWriter) error {
 			if initError := spawnInitContainer(client, ctx, containerName, mountList, importConfig, dog, cfg); initError != nil {
-				log.Printf("Failed to spawn init container: %v", initError)
+				errStr := fmt.Sprintf("Failed to spawn init container: %v", initError)
```
Comment (polaroi8d) [inline agent/pkg/dagent/utils/docker.go:409]: error start with lowercase

File: agent/pkg/dagent/utils/docker.go (hunk 1 above)
Comment (robot9706) [inline agent/pkg/dagent/utils/docker.go:410]: Dogger prints messages to stdout too, so this is redundant.

**Review by robot9706 — CHANGES_REQUESTED** [review]

File: agent/pkg/dagent/utils/docker.go (hunk 2 above)
Comment (nandor-magyar) [reply agent/pkg/dagent/utils/docker.go:409]: This contains an error, but it an info, completely intentional.

File: agent/pkg/dagent/utils/docker.go (hunk 1 above)
Comment (nandor-magyar) [reply agent/pkg/dagent/utils/docker.go:410]: Yes, for info, if something goes ham, letting that be visible.

**Review by polaroi8d — APPROVED** [review]
LGTM

**Pull request merged**